external/java/src/test/java/ch/epfl/dedis/lib/crypto/BdnSigTest.java

```diff
@@ -0,0 +1,80 @@
+package ch.epfl.dedis.lib.crypto;
+
+import ch.epfl.dedis.lib.Hex;
+import org.junit.jupiter.api.Test;
+
+import java.math.BigInteger;
+import java.security.SecureRandom;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.Random;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+class BdnSigTest {
+    private Random rnd = new SecureRandom();
+
+    @Test
+    void testHashPointToRReferences() {
+        Bn256G2Point p1 = new Bn256G2Point(BigInteger.ONE);
+        Bn256G2Point p2 = new Bn256G2Point(BigInteger.valueOf(2));
+        Bn256G2Point p3 = new Bn256G2Point(BigInteger.valueOf(3));
+        List<Point> pubs = Arrays.asList(p1, p2, p3);
+
+        List<Scalar> coefs = BdnSig.hashPointToR(pubs);
+
+        assertEquals("35b5b395f58aba3b192fb7e1e5f2abd3", Hex.printHexBinary(coefs.get(0).toBytes()).toLowerCase());
+        assertEquals("14dcc79d46b09b93075266e47cd4b19e", Hex.printHexBinary(coefs.get(1).toBytes()).toLowerCase());
+        assertEquals("933f6013eb3f654f9489d6d45ad04eaf", Hex.printHexBinary(coefs.get(2).toBytes()).toLowerCase());
```

Comment:
are these from a reference implementation or is it from kyber?

Reply:
From Kyber, there is a test that generates those values specifically for that purpose.

Reply:
There is no real reference implementation for this, they're quite obscure on the hash function to use.